### PR TITLE
febio-studio: init at 1.6.1

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/febio-studio/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ lib, stdenv, fetchFromGitHub, cmake, gcc, qt515, zlib, libGL, libGLU, openssl, libssh, xorg, wrapQtAppsHook, qtbase}:
```
Comment (ajs124) [inline pkgs/applications/science/biology/febio-studio/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, gcc, qt515, zlib, libGL, libGLU, openssl, libssh, xorg, wrapQtAppsHook, qtbase }:
```

File: pkgs/applications/science/biology/febio-studio/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "FEBioStudio";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "febiosoftware";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:0r6pg49i0q9idp7pjymj7mlxd63qjvmfvg0l7fmx87y1yd2hfw4h";
```
Comment (ajs124) [inline pkgs/applications/science/biology/febio-studio/default.nix:11]:
```suggestion
    sha256 = "0r6pg49i0q9idp7pjymj7mlxd63qjvmfvg0l7fmx87y1yd2hfw4h";
```

File: pkgs/applications/science/biology/febio-studio/default.nix (hunk 1 above)
Comment (ajs124) [reply pkgs/applications/science/biology/febio-studio/default.nix:1]: is this specific version of qt needed? `qt515` also seems to be the default, so why not just `qt5`?

File: pkgs/applications/science/biology/febio-studio/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  ];
+
+  cmakeFlags = [
+    "-DQt_Root=${qt515.full}"
+    "-DQt_Ver=5"
+    "-DNOT_FIRST=On"
+    "-DUSE_SSH=On"
+    "-DCAD_FEATURES=Off"
+    "-DMODEL_REPO=Off"
+    "-DUSE_FFMPEG=Off"
```
Comment (ajs124) [inline pkgs/applications/science/biology/febio-studio/default.nix:24]: why are all these things turned off? does a regular user of this program not need them?

File: pkgs/applications/science/biology/febio-studio/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "-DUSE_MMG=Off"
+    "-DUSE_TETGEN=Off"
+  ];
+
+  installPhase = ''
+    mkdir -p $out/
+    cp -R bin $out/
+  '';
+
+  nativeBuildInputs = [ cmake gcc wrapQtAppsHook ];
```
Comment (ajs124) [inline pkgs/applications/science/biology/febio-studio/default.nix:34]: doesn't `wrapQtAppsHook` get passed in automatically when using `libsForQt5.callPackage`?

File: pkgs/applications/science/biology/febio-studio/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  meta = {
+    description = "FEBio Suite Solver";
+    license = with lib.licenses; [ mit ];
+    homepage = "https://febio.org/";
+    platforms = lib.platforms.unix;
+    maintainers = with lib.maintainers; [ Scriptkiddi ];
+  };
+}
+
```
Comment (ajs124) [inline pkgs/applications/science/biology/febio-studio/default.nix:45]: these empty lines should be omitted

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -5429,6 +5429,8 @@ with pkgs;
 
   fbvnc = callPackage ../tools/admin/fbvnc {};
 
+  febio-studio = libsForQt5.callPackage ../applications/science/biology/febio-studio {};
```
Comment (ajs124) [inline pkgs/top-level/all-packages.nix:5432]: this is probably in the wrong section. there's one for `SCIENCE/BIOLOGY`, but `all-packages.nix` is a giant mess anyways -.-

**Review by ajs124 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/febio-studio/default.nix (hunk 1 above)
Comment (Scriptkiddi) [reply pkgs/applications/science/biology/febio-studio/default.nix:1]: It is the recommended version from the developers

File: pkgs/applications/science/biology/febio-studio/default.nix (hunk 4 above)
Comment (Scriptkiddi) [reply pkgs/applications/science/biology/febio-studio/default.nix:34]: not really sure, I took this from https://nixos.wiki/wiki/Qt#Packaging

File: pkgs/applications/science/biology/febio-studio/default.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:0r6pg49i0q9idp7pjymj7mlxd63qjvmfvg0l7fmx87y1yd2hfw4h";
+  };
+  patches = [
+    ./febio-studio-cmake.patch
+  ];
+
+  cmakeFlags = [
+    "-DQt_Root=${qt515.full}"
```
[the remark this answers is not in the document]
Comment (Scriptkiddi) [reply pkgs/applications/science/biology/febio-studio/default.nix:18]: it all has to be in one folder for the cmake file to works, i figured that would be the easiest

**Review by ajs124 — APPROVED** [review]

**Pull request merged**